File: BlackSmith/ClientApp/app/model/production.ts
import { ModelBase } from "./modelbase";
import { InventoryItem } from "./inventoryitem";
import { Product } from "./product";

export class Production extends ModelBase {
    date: string;
    productionProducts: ProductionProduct[];
    productionInventoryItems: ProductionInventoryItem[];
    isGenerated: boolean = false;
}

export class ProductionProduct extends ModelBase {
    quantity: number;
    detailNo: number;
    fK_ProductionId: number;
    fK_ProductId: number;
    product: Product;
}

export class ProductionInventoryItem extends ModelBase {
    quantity: number;
    detailNo: number;
    fK_ProductionId: number;
    fK_InventoryItemId: number;
    inventoryitem: InventoryItem;
    availableQuantity: number;
}

export class ProductionLoader extends ModelBase {
    items: ProductionInventoryItem[];
    products: ProductionProduct[];
}

export class ProductionList extends ModelBase {
    productions: Production[] = [];
}

File: BlackSmithViewModels/Production.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ViewModels
{
    public class Production : BaseModel
    {
        public virtual DateTime Date { get; set; }
        public virtual string Note { get; set; }
        public virtual List<ProductionInventoryItem> ProductionInventoryItems { get; set; }
        public virtual List<ProductionProduct> ProductionProducts { get; set; }

    }

    public class ProductionInventoryItem : BaseModel
    {
        public virtual long FK_ProductionId { get; set; }
        public virtual long FK_InventoryItemId { get; set; }
        public virtual decimal Quantity { get; set; }
        public virtual Production Production { get; set; }
        public virtual InventoryItem InventoryItem { get; set; }

        public virtual int DetailNo { get; set; }
    }

    public class ProductionProduct : BaseModel
    {
        public virtual long FK_ProductionId { get; set; }
        public virtual long FK_ProductId { get; set; }
        public virtual decimal Quantity { get; set; }
        public virtual Production Production { get; set; }
        public virtual Product Product { get; set; }
        public virtual int DetailNo { get; set; }

    }

    public class ProductionList : BaseModel
    {
        public virtual List<Production> Productions { get; set; }
    }
}